frontend/src/views/Collection/index.tsx

```diff
@@ -44,14 +50,43 @@ const Collection: FC = () => {
     ? VISIBILITY_TYPE.PRIVATE
     : VISIBILITY_TYPE.PUBLIC;
 
+  const [uploadLink] = useCollectionUploadLinks(id, visibility);
+
+  const [uploadedFiles, setUploadedFiles] = useState({} as UploadedFiles);
+
   const { data: collection, isError } = useCollection({ id, visibility });
 
   const [selectedTab, setSelectedTab] = useState(TABS.DATASETS);
 
+  const queryCache = useQueryCache();
+
   if (!collection || isError) {
     return null;
   }
 
+  const addNewFile = (newFile: UploadingFile) => {
```

Comment:
Collection page now owns the upload dataset state, because the add dataset button is now at the upper right corner